include/swift/Frontend/FrontendOptions.h

```diff
@@ -162,15 +158,15 @@ class FrontendInputs {
                : Optional<StringRef>();
   }
 
-  bool isPrimaryInputAFileAt(unsigned i) const {
+  bool isThereAPrimaryInputWithAFilenameAt(unsigned i) const {
```

Comment:
This still doesn't make sense to me. Why is this using "a" at all?

Reply:
Do you like "isInputPrimary" better? Can't recall exactly why it doesn't make sense.

Reply:
That is better, yes. `isPrimaryInputAFileAt` is weird because it's asking two questions: "is the primary input a file" and "is the primary input at `i`", but "is the primary input a file at `i`" doesn't mean that. `isThereAPrimaryInputWithAFilenameAt` is better, but seems overly circuitous when you could just ask "is the input at `i` primary (and is it a file)".

It sounds like your *next* patch defines this whole thing away because (a) there are no non-file inputs, and (b) the inputs themselves know whether they are primary. So maybe it's not worth changing right now.

Reply:
In general, our style omits "a"/"an"/"the", so if you're including it I want it to be significant.